classes/class-wpcom-liveblog-entry.php

```diff
@@ -358,45 +354,56 @@ public static function handle_restricted_shortcodes( $args ) {
 	}
 
 	/**
-	 * If author select is enabled return the user using author_id,
-	 * if user not found then set as current user as a fallback.
+	 * Return the user using author_id, if user not found then set as current
+	 * user as a fallback, we store a meta to show that authors are hidden as
+	 * a comment must have an author.
 	 *
 	 * If a entry_id is supplied we should update it as its the
 	 * original entry which is used for displaying author information.
 	 *
+	 *
 	 * @param array $args The new Live blog Entry.
 	 * @param int   $entry_id If set we should update the original entry
 	 * @return mixed
 	 */
 	private static function handle_author_select( $args, $entry_id ) {
 		if ( isset( $args['author_id'] ) && $args['author_id'] ) {
-			$user_object = get_userdata( $args['author_id'] );
+			$user_object = self::get_userdata_with_filter( $args['author_id'] );
 			if ( $user_object ) {
 				$args['user'] = $user_object;
 
-				if ( $entry_id ) {
```

Comment:
What's the reason for removing the conditional here? Looks like this is probably to protect against errors if `$entry_id` isn't set.

Reply:
@philipjohn I believe the if statement was removed due to it being redundant as all calls to the function make a check to see if the `$editor_id` parameter is set and not empty prior to calling this function; in addition to this the aforementioned if statement is wrapped within another if statement meaning if the earlier if fails then it resorts to using the `else` block.

As a solution we could superfluously add this if statement back but at the top of this function block which will just return `$args['user'];` to prevent the rest of the code running.